storey/aggregations.py

```diff
@@ -248,15 +248,16 @@ async def _do(self, event):
             self._terminate_worker = True
             return await self._do_downstream(_termination_obj)
 
-        try:
-            element = event.body
-            key = event.key
-            if self.key_extractor:
+        element = event.body
+        key = event.key
+        if self.key_extractor:
+            try:
                 key = self.key_extractor(element)
-            await self._emit_event(key, event)
-
-        except Exception as ex:
-            raise ex
+            except Exception:
+                # in case of an error, emit event without key and features
+                await self._do_downstream(event)
+                return
+        await self._emit_event(key, event)
```

Comment:
Need to use `get` instead of [] to avoid the exception being thrown in the first place.